notes/lecture22.tex

```diff
@@ -0,0 +1,205 @@
+
+
+\section{Lecture 22: Non-convex constraints} 
+
+\begin{center}
+Seyed Mehran Mirramezani and Serena Yuan
+\end{center}
+
+
+Last time we discussed optimization of a convex function $f$ over non-convex sets. A typical example for such a problem is $\min \| Ax-y\|_2^2$ subject to $\|x\|_0 \leq s$. Another type of non-convex optimization is to minimize a non-convex function over a convex set. A typical example for this kind of non-convex optimization is known as $l_0$-minimization where $\min \| x\|_0$ subject to an affine convex constraint like $Ax=y$. In sum, the $l_0$-minimization is: 
+
+$$\min_{Ax=y} \|x\|_0$$
+
+One option for solving the above non-convex optimization problem is to relax the $l_0$-objective to the convex $l_1$-objective as:
+
+$$\min_{Ax=y} \|x\|_1$$
+
+A variation of the above problem is Basis Pursuit Denoising (BPDN),
+
+\begin{align}
+\min \|x\|_1 \text{ s.t. } \|Ax - y \|_2 \leq \epsilon.
+\end{align}
+
+
+Note that under appropriate assumptions on $A$, $x$ and $y$, we observed that $l_1$-minimization still gives the right answer (RIP, restricted nullspace).
+Sparse linear regression has received a lot of attention and there are many related optimization programs like:
+BPDN (allows for noise)
+$$\min_{\|Ax-y\|_2 \leq s} \|x\|_1$$
+
+Constrained Lasso
+$$\min_{\|x\|_1 \leq \lambda }\|Ax-y\|_2^2 $$
+
+Lagrangian Lasso
+$$\min \|Ax-y\|_2^2 +  \lambda \|x\|_1$$
+
+In our setting, we have \\
+
+\centering{nonconvex constraint} 
+$$\downarrow$$
+\centering{cvx relaxation }
+$$\downarrow$$
+\centering{PGD (Projected Gradient Descent)},
+
+or more directly, nonconvex constraint $\rightarrow$ PGD. 
+
+Here, the Restricted Isometry Property allows optimization over nonconvex sets.
+
+
+Now, we discuss the Iterative Hard Thresholding (IHT) which is known as a projected gradient descent (PGD) for sparse vectors.
+
+Iterative Hard Thresholding (IHT) is used with Gradient Descent for the following problem,
```

Comment:
Note that the next block of equations is not defining a problem, only an objective function. It would be good to explicitly state the full problem.

Reply:
I tried to look up the full problem. Is this a good description?

Given the setup
\begin{align}
y = \Phi x + e 
\end{align}

where $y \in \R^M, x \in \R^N, \Phi \in \R^{M \times N}$, $e$ is observation noise, to estimate $x$ given $y$ and $\Phi$ when $M <<N$ and $x$ is approximately $K$-sparse.

The IHT algorithm uses the iteration
\begin{align}
x^{n+1} = P_K ( x^n + \Phi^{\trans} (y - \Phi x^n))
\end{align}

where $P_K$ is a hard thresholding operator that keeps the largest $K$ elements of a vector.

Reply:
Yes, this is a good description. But we should use the same notation as in the rest of the note (e.g., A instead of Phi) and make sure there is not too much overlap with the "setup" section below (the problem statement here and the setup can probably be merged at least partially).